### ISSUE TITLE
[rust_port] Gtk-warning markup error

### ISSUE THREAD

**Issue by sigmaSd** [haecker-felix/Gradio#336]
``
(Radio:30431): Gtk-WARNING **: 16:08:48.933: Failed to set text '<a href="https://www.internet-radio.com/servers/tools/playlistgenerator/?u=http://184.173.185.71:9804/listen.pls&t=.m3u">https://www.internet-radio.com/servers/tools/playlistgenerator/?u=http://184.173.185.71:9804/listen.pls&t=.m3u</a>' from markup due to error parsing markup: Error on line1: Entity did not end with a semicolon; most likely you used an ampersand character without intending to start an entity - escape ampersand as &amp;
``

Wasn't sure what to about it,cause I get this error with the same station so you could just set the homepage of stations that include ``&`` to ``""`` or handle it and add ``amp;`` to escape this character.

**Comment by sigmaSd** [comment]
Hmm now I understand , basically I get this error when a station have the wrong homepage (live stream url instead of the actual homepage),
not sure how do you want to handle it.

**Comment by haecker-felix** [comment]
No. The message appears, if the url contains a special character.

For example the German 'Ä' letter needs to be translated into "&auml". The error message itself says it:

"most likely you used an ampersand character without intending to start an entity - escape ampersand as &amp;"

**Comment by sigmaSd** [comment]
that's pretty cool , didn't know that , I added a PR to fix it , I hope it's not too hacky .
